arxivdigest/api/database.py

```diff
@@ -300,22 +323,28 @@ def get_user_feedback_topics(ids):
             {topic: {
                 'seen': date or None,
                 'clicked': date or None,
+                'state': varchar or None,
                 }, ...
             }, ...
         ], ...
     }
     """
     with closing(getDb().cursor(dictionary=True)) as cur:
-        sql = """SELECT user_id, topic, DATE(datestamp) as date, seen, clicked
-                 FROM topic_recommendations tr NATURAL JOIN topics t
-                 WHERE user_id IN ({}) AND interleaving_order IS NOT 
-                 null ORDER BY  interleaving_order 
-                 DESC""".format(','.join(['%s'] * len(ids)))
+        sql = '''SELECT tr.user_id, t.topic, DATE(tr.datestamp) as date, 
```

Comment:
This function should also return data/feedback on user added topics, such that systems does not recommend them again.